tracer/src/Datadog.Trace/AppSec/SecuritySettings.cs

```diff
@@ -85,39 +85,36 @@ bool GetEnabledDefaultValue()
                                                || val.Equals(UserTrackingExtendedMode, StringComparison.OrdinalIgnoreCase))
                                          .ToLowerInvariant();
 
-            ApiSecuritySampling = config
-                                 .WithKeys(ConfigurationKeys.AppSec.ApiSecurityRequestSampleRate)
-                                 .AsDouble(defaultValue: 0.1, validator: val => val is <= 1 and >= 0)
-                                 .Value;
-
-            ApiSecurityMaxConcurrentRequests = config
-                                              .WithKeys(ConfigurationKeys.AppSec.ApiSecurityMaxConcurrentRequests)
-                                              .AsInt32(defaultValue: 1, validator: val => val >= 1)
-                                              .Value;
-
             ApiSecurityEnabled = config.WithKeys(ConfigurationKeys.AppSec.ApiExperimentalSecurityEnabled)
                                        .AsBool(false);
+
+            ApiSecuritySampleDelay = config.WithKeys(ConfigurationKeys.AppSec.ApiSecuritySampleDelay)
+                                           .AsDouble(30.0, val => val >= 0.0)
+                                           .Value;
+
             UseUnsafeEncoder = config.WithKeys(ConfigurationKeys.AppSec.UseUnsafeEncoder)
                                      .AsBool(false);
 
             // For now, RASP is disabled by default.
             RaspEnabled = config.WithKeys(ConfigurationKeys.AppSec.RaspEnabled)
-                                     .AsBool(false) && Enabled;
+                                .AsBool(false) && Enabled;
 
             StackTraceEnabled = config.WithKeys(ConfigurationKeys.AppSec.StackTraceEnabled)
-                         .AsBool(true);
+                                      .AsBool(true);
 
             MaxStackTraces = config
-                                  .WithKeys(ConfigurationKeys.AppSec.MaxStackTraces)
-                                  .AsInt32(defaultValue: 2, validator: val => val >= 0)
-                                  .Value;
+                            .WithKeys(ConfigurationKeys.AppSec.MaxStackTraces)
+                            .AsInt32(val => val >= 0)
+                            .GetValueOrDefault(2);
 
             MaxStackTraceDepth = config
-                                  .WithKeys(ConfigurationKeys.AppSec.MaxStackTraceDepth)
-                                  .AsInt32(defaultValue: 32, validator: val => val >= 0)
-                                  .Value;
+                                .WithKeys(ConfigurationKeys.AppSec.MaxStackTraceDepth)
+                                .AsInt32(val => val >= 0)
+                                .GetValueOrDefault(32);
```

Comment:
You shouldn't use `.GetValueOrDefault` there's a default parameter in the `.AsInt32`, it means the collection system is aware of the default.

Reply:
ah yes bad rebase, fixed in [172591f](https://github.com/DataDog/dd-trace-dotnet/pull/5257/commits/172591fe4a0f50752d8aa02ab7aeecbcdf3180a5)